### PR TITLE
Changes related to setting VpnClient AAD authentication options to Virtual network gateway resource & Advertising virtual Wan 2 released APIs.

### PR DESCRIPTION
**Pull request by Nilambari**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualWan.json [hunk 1]
```diff
@@ -2300,7 +2349,56 @@
           "final-state-via": "location"
         }
       }
-    }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/p2svpnGateways/{gatewayName}/getp2svpnconnectionhealth": {
```
Comment (nschonni) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualWan.json]:
```suggestion
    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/p2svpnGateways/{gatewayName}/getP2sVpnConnectionHealth": {
```

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/examples/P2SVpnGatewayGetConnectionHealth.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "parameters": {
+    "gatewayName": "p2sVpnGateway1",
+    "resourceGroupName": "rg1",
+    "api-version": "2019-04-01",
+    "subscriptionId": "subid"
+  },
+  "responses": {
+    "200": {
```
Comment (EvgenyAgafonchikov) [inline specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/examples/P2SVpnGatewayGetConnectionHealth.json]: Please also add entry for 202 response code

**Review by EvgenyAgafonchikov — CHANGES_REQUESTED** [review]
Left some comments, please take a look

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/examples/P2SVpnGatewayGetConnectionHealth.json (hunk 2 above)
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/examples/P2SVpnGatewayGetConnectionHealth.json]: Done.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/examples/VirtualNetworkGatewayGet.json [hunk 3]
```diff
@@ -42,6 +42,16 @@
                     "vpnType": "RouteBased",
                     "enableBgp": false,
                     "activeActive": false,
+                    "vpnClientConfiguration": {
+                        "vpnClientProtocols": [
+                            "OpenVPN"
+                        ],
+                        "vpnClientRootCertificates": [],
+                        "vpnClientRevokedCertificates": [],
+						"aadTenant": "https://login.microsoftonline.com/99163270-c9ab-4279-bb12-5353ed9bf6e1",
```
[the remark this answers is not in the document]
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/examples/VirtualNetworkGatewayGet.json]: Done.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/examples/VirtualNetworkGatewayGet.json [hunk 4]
```diff
@@ -42,6 +42,16 @@
...
+						"aadAudience": "fd11706c-e78c-4828-8be7-ac7dcefd4b17",
+						"aadIssuer": "https://sts.windows.net/99163270-c9ab-4279-bb12-5353ed9bf6e1/"				
+                    },
```
[the remark this answers is not in the document]
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/examples/VirtualNetworkGatewayGet.json]: Done.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualWan.json [hunk 5]
```diff
@@ -1390,6 +1390,55 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/vpnGateways/{gatewayName}/reset": {
+      "post": {
+        "tags": [
+          "vpnGateways"
```
[the remark this answers is not in the document]
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualWan.json]: Done.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualWan.json [hunk 6]
```diff
@@ -1390,6 +1390,55 @@
...
+        ],
+        "operationId": "vpnGateways_Reset",
```
[the remark this answers is not in the document]
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualWan.json]: Done.

File: specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualWan.json (hunk 1 above)
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2019-04-01/virtualWan.json]: Done.

**Comment by Nilambari** [comment]
Incorporated review comments. Please take a look. Thanks!

**Review by EvgenyAgafonchikov — APPROVED** [review]

**Comment by MikhailTryakhov** [comment]
@sergey-shandar  can you please take a look?

**Comment by sergey-shandar** [comment]
@Nilambari, @MikhailTryakhov, @EvgenyAgafonchikov it still fails in Network CI. Should we ignore this errors?

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**